pkg/providers/imagefamily/bootstrap/aksbootstrap.go

```diff
@@ -145,32 +145,19 @@ var (
 	// NodeTemplate                : user input that could be per-node (t) - ALL DEFAULTED FOR NOW
 	// Provisioner spec            : selected nodepool-level user input (p)
 
-	// NodeBootstrapVariables carries all variables needed to bootstrap a node
+	// staticNodeBootstrapVars carries all variables needed to bootstrap a node
 	// It is used as input rendering the bootstrap script Go template (customDataTemplate)
 	// baseline, covering unused (-), static (s), and unsupported (n) fields,
 	// as well as defaults, cluster/node level (cd/td/xd)
 	staticNodeBootstrapVars = nbcontractv1.Configuration{
-		CustomCloudConfig: &nbcontractv1.CustomCloudConfig{
-			IsAksCustomCloud:  false,              //n
-			InitFilePath:      "",                 //n
-			RepoDepotEndpoint: "",                 //n
-			TargetEnvironment: "AzurePublicCloud", //n
-		},
-		LinuxAdminUsername: "azureuser", // td
-		KubeBinaryConfig: &nbcontractv1.KubeBinaryConfig{
-			KubeBinaryUrl:        "", // cd
-			CustomKubeBinaryUrl:  "", // -
-			PrivateKubeBinaryUrl: "",
-		},
-		KubeproxyUrl: "", // -
-		ApiserverConfig: &nbcontractv1.ApiServerConfig{
+		CustomCloudConfig: &nbcontractv1.CustomCloudConfig{},
+		KubeBinaryConfig:  &nbcontractv1.KubeBinaryConfig{},
```

Comment:
for the configs that will be used later on, im wondering if its better to initialize it there where we know we are using it. e.g. in `applyOptions`, right be we set the kubebinaryconfig fields, we do a check like:
```
if nbv.KubeBinaryConfig == nil {
// intialize KubeBinaryConfig
}
// set rest of fields
```

Reply:
Let me clarify a bit what you meant with 2 examples.
1. LinuxAdminUsername: if this is not set, it will be set to "azureuser" when executing the go template. For this case, we don't do anything in `applyOptions`, right? And if Karpenter wants to set a custom user, do you mean setting it in `applyOptions` instead of in `staticNodeBootstrapVars` declaration?
2. KubeBinaryConfig: do you mean we don't initialize it in `staticNodeBootstrapVars` declaration, but doing it (assigning the actual values) in `applyOptions`?

Reply:
Yes to both. What do you think? Would it be better to check for nil and then intitalize in `applyOptions`. I think this might be safer since you are ensuring it is initlaized before you access it. Though it might look messier.

Reply:
I said too early. It's not 35 because some others are in `applyOptions`, though some of them can be computed.